core/src/main/java/com/ritense/valtimo/service/CamundaTaskService.java

```diff
@@ -366,34 +371,40 @@ public Page<TaskExtended> findTasksFiltered(
         var assigneeMap = new java.util.HashMap<String, ValtimoUser>();
         var tasks = typedQuery.getResultList().stream()
             .map(tuple -> {
-                var task = tuple.get(0, CamundaTask.class);
-                var executionId = tuple.get(1, String.class);
-                var businessKey = tuple.get(2, String.class);
-                var processDefinitionId = tuple.get(3, String.class);
-                var processDefinitionKey = tuple.get(4, String.class);
-
-                ValtimoUser valtimoUser;
-                if (task.getAssignee() == null) {
-                    valtimoUser = null;
-                } else if (assigneeMap.containsKey(task.getAssignee())) {
-                    valtimoUser = assigneeMap.get(task.getAssignee());
-                } else {
-                    valtimoUser = getValtimoUser(task.getAssignee());
-                    assigneeMap.put(task.getAssignee(), valtimoUser);
+                try {
+                    var task = tuple.get(0, CamundaTask.class);
+                    var executionId = tuple.get(1, String.class);
+                    var businessKey = tuple.get(2, String.class);
+                    var processDefinitionId = tuple.get(3, String.class);
+                    var processDefinitionKey = tuple.get(4, String.class);
+
+                    ValtimoUser valtimoUser;
+                    if (task.getAssignee() == null) {
+                        valtimoUser = null;
+                    } else if (assigneeMap.containsKey(task.getAssignee())) {
+                        valtimoUser = assigneeMap.get(task.getAssignee());
+                    } else {
+                        valtimoUser = getValtimoUser(task.getAssignee());
+                        assigneeMap.put(task.getAssignee(), valtimoUser);
+                    }
+
+                    var context = task.getVariable(CONTEXT);
+
+                    return TaskExtended.of(
+                        task,
+                        executionId,
+                        businessKey,
+                        processDefinitionId,
+                        processDefinitionKey,
+                        valtimoUser,
+                        context
+                    );
+                } catch(Exception e) {
+                    logger.error("Failed to build task "+tuple, e);
+                    return null;
```

Comment:
Rather than filtering out the results, I think it makes more sense to update those records so the users are asynchronously unassigned from the relevant tasks. Additionally, the user should just be excluded from the results, but the task itself should still be returned.

Reply:
It does not seem like a good idea to us to remove the users from the tasks, simply because you tried to retrieve a list of tasks. This would mean throwing away data that could not be retrieved anymore, and should be done with some consideration, not as a side effect.

For instance, a bug where the migration from userEmail to userId fails, would end up in wiping all the users from all the tasks.